latex/epoch.tex

```diff
@@ -1,54 +1,614 @@
-The pool reaping in rules are discribed in
-Figure~\ref{fig:ts-types:pool-reap} and Figure~\ref{fig:rules:pool-reap}.
+\newcommand{\UTxOEpEnv}{\type{UTxOEpEnv}}
+\newcommand{\UTxOEpState}{\type{UTxOEpState}}
+\newcommand{\Accnt}{\type{Accnt}}
+\newcommand{\AccntEnv}{\type{AccntEnv}}
+\newcommand{\AccntState}{\type{AccntState}}
+\newcommand{\StPlCleanEnv}{\type{StPlCleanEnv}}
+\newcommand{\StPlCleanState}{\type{StPlCleanState}}
+\newcommand{\NewProtoConstsEnv}{\type{NewProtoConstsEnv}}
+\newcommand{\NewProtoConstsState}{\type{NewProtoConstsState}}
+\newcommand{\EpochEnv}{\type{EpochEnv}}
+\newcommand{\EpochState}{\type{EpochState}}
+\newcommand{\Production}{\type{Production}}
+
+\newcommand{\obligation}[4]{\fun{obligation}~ \var{#1}~ \var{#2}~ \var{#3}~ \var{#4}}
+\newcommand{\reward}[5]{\fun{reward}~ \var{#1}~ \var{#2}~ \var{#3}~ \var{#4}~ \var{#5}}
+\newcommand{\poolRefunds}[3]{\fun{poolRefunds}~ \var{#1}~ \var{#2}~ \var{#3}}
 
 %%
-%% Figure - Pool Reap
+%% Figure - Epoch Abstract Types
 %%
+\begin{figure}
+  \emph{Abstract types}
+  %
+  \begin{equation*}
+    \begin{array}{r@{~\in~}lr}
+      \var{production} & \Production & \text{blocks produced in the epoch}\\
+    \end{array}
+  \end{equation*}
+  \caption{Epoch definitions}
+  \label{fig:epoch-defs}
+\end{figure}
+
+%%
+%% Figure - Functions for Epoch Rules
+%%
+\begin{figure}
+  \begin{align*}
+      & \fun{obligation} \in \PrtclConsts \to \Allocs \to \Allocs \to \Slot \to \Coin
+      & \text{total possible refunds} \\
+      & \obligation{pc}{stkeys}{stpools}{cslot} =\\
+      & \sum\limits_{(\_ \mapsto s) \in \var{stkeys}}
+        \refund{k_{\mathsf{val}}}{k_{\min}}{(\slotminus{cslot}{s})}{\lambda} \\
+      & + \sum\limits_{(\_ \mapsto s) \in \var{stpools}}
+        \refund{p_{\mathsf{val}}}{p_{\min}}{(\slotminus{cslot}{s})}{\lambda} \\
+      &
+        \where k_{\mathsf{val}}, k_{\min},p_{\mathsf{val}}, p_{\min}, \lambda \in pc
```

Comment:
I define `pc` to be a map from "atoms" a la lisp (or just keys) to values, so the where clause would read something like `pcKVal \mapsto k_val \in pc`. Maybe there's a better notation, but by `someCst \in pc` we could make a replacement of variables and have `x \in pc` with doesn't tell us much :)

Reply:
that sounds great

Reply:
I went for just another accessor, since that was more consistent with everything else.  But I'm definitely still looking to clean things up, and changing to a map for these constants may be the best way to go.